src/helpers/dimensions.ts

```diff
@@ -0,0 +1,166 @@
+import { MEASUREGROUP } from '../constants/dimensions';
+import {
+    VIEW,
+    STACK,
+    SEGMENT,
+    TREND,
+    ATTRIBUTE,
+    MEASURES
+ } from '../constants/bucketNames';
+
+function getDimensionTotals(bucket: VisualizationObject.IBucket): AFM.ITotalItem[] {
+    const bucketTotals = get(bucket, 'totals', []);
+    return bucketTotals.map((total: VisualizationObject.IVisualizationTotal): AFM.ITotalItem => {
+        return {
+            measureIdentifier: total.measureIdentifier,
+            type: total.type,
+            attributeIdentifier: total.attributeIdentifier
+        };
+    });
+}
+
+function getTableDimensions(mdObject: VisualizationObject.IVisualizationObjectContent): AFM.IDimension[] {
+    const attributes: VisualizationObject.IBucket = mdObject.buckets
+        .find(bucket => bucket.localIdentifier === ATTRIBUTE);
+
+    const measures: VisualizationObject.IBucket = mdObject.buckets
+        .find(bucket => bucket.localIdentifier === MEASURES);
+
+    const attributesItemIdentifiers = get(attributes, 'items', [])
+        .map((a: VisualizationObject.IVisualizationAttribute) =>
+            a.visualizationAttribute.localIdentifier);
+
+    const measuresItemIdentifiers =
+        measures && measures.items && measures.items.length ? [MEASUREGROUP] : [];
+
+    const totals = getDimensionTotals(attributes);
+    const totalsProp = totals.length ? { totals } : {};
+
+    return [{
+        itemIdentifiers: attributesItemIdentifiers,
+        ...totalsProp
+    }, {
+        itemIdentifiers: measuresItemIdentifiers
+    }];
+}
+
+function getPieDimensions(mdObject: VisualizationObject.IVisualizationObjectContent): AFM.IDimension[] {
+    const view = mdObject.buckets.find(bucket => bucket.localIdentifier === VIEW);
+
+    if (view && view.items.length) {
+        return [
+            {
+                itemIdentifiers: [MEASUREGROUP]
+            },
+            {
+                itemIdentifiers: view.items.map((attribute: VisualizationObject.IVisualizationAttribute) =>
+                    attribute.visualizationAttribute.localIdentifier)
+            }
+        ];
+    }
+
+    return [
+        {
+            itemIdentifiers: []
+        },
+        {
+            itemIdentifiers: [MEASUREGROUP]
+        }
+    ];
+}
+
+function getBarDimensions(mdObject: VisualizationObject.IVisualizationObjectContent): AFM.IDimension[] {
+    const view: VisualizationObject.IBucket = mdObject.buckets
+            .find(bucket => bucket.localIdentifier === VIEW);
+
+    const stack: VisualizationObject.IBucket = mdObject.buckets
+        .find(bucket => bucket.localIdentifier === STACK);
+
+    const hasNoStacks = !stack || !stack.items || stack.items.length === 0;
+
+    if (hasNoStacks) {
+        return [
+            {
+                itemIdentifiers: [MEASUREGROUP]
+            },
+            {
+                itemIdentifiers: (view && view.items || [])
```

Comment:
too much code in one line. Please if its possible to split it... 
`v` shortcut looks bad as well

Reply:
fixed by moving code to named function